### PR TITLE
JDK-8287674: CSS improvements for summary lists

### PR DESCRIPTION
**Pull request by hns**
This is a refinement of CSS styles mostly in the deprecated and new API pages.

 - Provide a two-row table view for the new sortable three-column tables
 - Make sure no line wraps are inserted between a checkbox and its label
 - Slightly increase the line height of the check box row to make it easier to use when wrapped
 - Define the overflow of summary table cells as "auto" which makes them scrollable if necessary

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.java.net/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8287674](https://bugs.openjdk.org/browse/JDK-8287674): CSS improvements for summary lists

### Reviewers
 * [Jonathan Gibbons](https://openjdk.java.net/census#jjg) (@jonathan-gibbons - **Reviewer**) ⚠️ Review applies to [d595fd7c](https://git.openjdk.java.net/jdk/pull/8975/files/d595fd7c122d3e9ded2d56aeafa491a4c2b5eb66)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8975/head:pull/8975` \
`$ git checkout pull/8975`

Update a local copy of the PR: \
`$ git checkout pull/8975` \
`$ git pull https://git.openjdk.java.net/jdk pull/8975/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8975`

View PR using the GUI difftool: \
`$ git pr show -t 8975`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8975.diff">https://git.openjdk.java.net/jdk/pull/8975.diff</a>

</details>

### CODE REVIEW

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/DeprecatedListWriter.java [hunk 1]
```diff
@@ -90,7 +91,7 @@ protected void addExtraSection(DeprecatedAPIListBuilder list, Content content) {
                         ? contents.getContent("doclet.Deprecated_API_Checkbox_Other_Releases")
                         : Text.of(release);
                 HtmlId htmlId = HtmlId.of("release-" + releaseIndex);
-                tabs.add(HtmlTree.LABEL(htmlId.name(),
+                tabs.add(new HtmlTree(TagName.WBR)).add(HtmlTree.LABEL(htmlId.name(),
```
Comment (jonathan-gibbons) [inline src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/DeprecatedListWriter.java:94]: Possible (future?) candidate for a static factory method.
Do we want to formally transition to always using static factory methods, and make the `new HtmlTree` constructors private?

**Review by jonathan-gibbons — APPROVED** [review]
Approved with suggestion for future.

**Comment by hns** [comment]
I have replaced the `<wbr>` elements between checkboxes with plain space characters as the former reduces the spacing between checkboxes on Chrome. With plain space characters, spacing is the same as before on all tested browsers and line wrapping works as expected.

**Pull request closed without merging**